light/client.go

```diff
@@ -974,9 +984,8 @@ func (c *Client) getLightBlock(ctx context.Context, p provider.Provider, height
 
 // NOTE: requires a providerMutex lock
 func (c *Client) removeWitnesses(indexes []int) error {
-	// check that we will still have witnesses remaining
-	if len(c.witnesses) <= len(indexes) {
-		return ErrNoWitnesses
+	if len(indexes) < 1 {
+		return nil
```

Comment:
I think the correct behavior here is still to return an error. If a user intended to run the light client with witnesses and thus wanted to cross check headers it would be preferable to them that the light client fails when it runs out of witnesses rather than simply continuing but now without cross checking any headers